Services/Catalogue/Mappers/ConfigurableProductsData.php

```diff
@@ -266,6 +277,30 @@ private function getCategoryArray($category)
             $parentUrlKey = '';
         }
 
+        if ($this->allStoreBaseUrls === null) {
+            $this->allStoreBaseUrls = $this->storeManager->getAllStoreBaseUrls();
+        }
+
+        $storeUrls = $this->allStoreBaseUrls;
+
+        if (!isset($this->currentStoreBaseUrl[$this->storeId])) {
+            $this->currentStoreBaseUrl[$this->storeId] = $this->storeManager->getCurrentStoreBaseUrl();
+            ;
+        }
+
+        $currentStoreBaseUrl = $this->currentStoreBaseUrl[$this->storeId];
```

Comment:
Check if 

$this->currentStoreBaseUrl[$this->storeId] and isset($this->currentStoreBaseUrl[$this->storeId]['base_url'])